torch/distributed/_sharded_tensor/api.py

```diff
@@ -0,0 +1,186 @@
+from typing import List
+
+import torch
+import torch.distributed as dist
+from torch.distributed._sharding_spec import (
+    ChunkShardingSpec,
+    ShardMetadata,
+    ShardingSpec,
+)
+from torch.distributed._sharding_spec._internals import is_valid_device
+from torch.distributed.utils import _parse_remote_device
+
+
+class ShardWithMetadata(object):
```

Comment:
Can't we just call this class `Shard`, and this class can have two fields called '_tensor' and `_metadata`? We can also even rename `ShardMetadata` as `Metadata` if there is not other metadata classes in the same package.